src/openrct2/world/Climate.cpp

```diff
@@ -181,7 +181,7 @@ extern "C"
         else if (gClimateCurrentWeatherEffect == 2)
         {
             // Create new thunder and lightning
-            uint32 randomNumber = util_rand();
+            uint32 randomNumber = scenario_rand();
```

Comment:
After sleeping over it I just realized that, altho it should still use the same prng stream however not affect it, that should at least have the same effects on all the clients.